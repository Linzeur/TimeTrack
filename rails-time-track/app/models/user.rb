class User < ApplicationRecord
  has_secure_password
  has_secure_token

  has_many :project_members
  has_many :projects, through: :project_members
  has_many :daily_logs, through: :project_members

  def self.valid_login?(email, password)
    user = find_by(email: email)
    user if user&.authenticate(password)
  end
end
