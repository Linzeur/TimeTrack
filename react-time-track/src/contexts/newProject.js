import React from "react";

const NewProjectContext = React.createContext();

function NewProjectProvider({ infoProject, children }) {
  return (
    <NewProjectContext.Provider value={infoProject}>
      {children}
    </NewProjectContext.Provider>
  );
}

function useNewProjectConsumer() {
  return React.useContext(NewProjectContext);
}

export { NewProjectProvider, useNewProjectConsumer };
