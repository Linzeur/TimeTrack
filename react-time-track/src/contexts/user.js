import React from "react";

const UserContext = React.createContext();

function UserProvider({ children }) {
  const [loggedIn, setLoggedIn] = React.useState(
    JSON.parse(localStorage.getItem("user")) || null
  );

  React.useEffect(() => {
    if (loggedIn) localStorage.setItem("user", JSON.stringify(loggedIn));
    else localStorage.removeItem("user");
  }, [loggedIn]);

  const onLogin = userData => setLoggedIn(userData);
  const onLogout = () => setLoggedIn(null);

  const value = {
    data: loggedIn,
    onLogin: onLogin,
    onLogout: onLogout
  };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
}

export { UserProvider, UserContext };
