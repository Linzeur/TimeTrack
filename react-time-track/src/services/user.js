import { apiUrl } from "../util";
// "http://localhost:3000/api/"

const API_USERS = `${apiUrl}users`;

async function userList() {
  const userList = await fetch(API_USERS, {
    method: "GET",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    }
  });

  if (!userList.ok) {
    const { errors } = await userList.json();
    console.log(errors);
    throw new Error(errors);
  }

  return userList.json();
}

async function createUser(userData) {
  console.log(userData);
  const response = await fetch(API_USERS, {
    method: "POST",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(userData)
  });

  if (!response.ok) {
    const { errors } = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
} 

async function editUser(userId) {
  console.log(userId);
  const response = await fetch(`${API_USERS}/${userId}`, {
    method: "PUT",
    credentials: "include",
    headers: {
      "Content-type": "application/json"
    }
  });

  if (!response.ok) {
    const {errors} = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
}

async function getUser(userId) {
  console.log(userId);
  const response = await fetch(`${API_USERS}/${userId}`, {
    method: "GET",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    }
  });

  if (!response.ok) {
    const { errors } = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
} 

export { userList, createUser, getUser, editUser };
