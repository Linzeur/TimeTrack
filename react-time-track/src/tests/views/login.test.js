import React from "react";
import { render } from "@testing-library/react";

import { UserProvider } from "../../contexts/user";
import Login from "../../views/login";

test("login component", () => {
  const user = {};
  const setUser = jest.fn();
  const { asFragment } = render(
    <UserProvider user={user} setUser={setUser}>
      <Login />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
