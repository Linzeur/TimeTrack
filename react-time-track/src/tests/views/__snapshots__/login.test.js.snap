// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component 1`] = `
<DocumentFragment>
  <div
    class="css-1s2t2fo-Login"
  >
    <form
      class="css-7zc9xd-Login"
    >
      <h1
        class="css-b55j7b-Login"
      >
        TimeTracker
      </h1>
      <label
        class="css-jidqlr-Login"
        for="Username"
      >
        Username
      </label>
      <input
        aria-label="Enter your email"
        autocomplete="off"
        class="css-ulpja2-Login"
        name="email"
        placeholder="Enter your email"
        required=""
        type="email"
        value=""
      />
      <label
        class="css-jidqlr-Login"
        for="Username"
      >
        Password
      </label>
      <input
        aria-label="Enter your password"
        autocomplete="off"
        class="css-ulpja2-Login"
        name="password"
        placeholder="Enter your password"
        required=""
        type="password"
        value=""
      />
      <br />
      <label
        class="css-dlvh0q-Login"
      >
        <a
          href="/"
        >
          Forgot your password?
        </a>
      </label>
      <button
        class="css-nxgp5n-Login"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
